bokehjs/src/coffee/models/glyphs/patches.coffee

```diff
@@ -77,8 +77,8 @@ class PatchesView extends Glyph.View
   _render: (ctx, indices, {sxs, sys}) ->
     # @sxss and @syss are used by _hit_point and sxc, syc
     # This is the earliest we can build them, and only build them once
```

Comment:
> Won't this get built every render call, not just once? 

https://github.com/bokeh/bokeh/commit/064004bdd64e339f939b055b29f6b35651c2ef70

Here's the commit that moved those lines and made that comment. The commit message is "Only build sxss and syss once - Instead of on every hit!"

So the comment at least makes sense now.
----
>  Maybe we could move to `_map_data`.

Maybe. I definitely haven't got my head deep into this.

> I'm also a little worried that since render is called with different sxs and sys for selection and non-selection glyphs

At the end of the day we only need to be building these sxss & syss once for the main piece of data as the point of it is so that hit testing can work for patches.

I haven't yet understood how selection/nonselection glyphs work and render.

So your concern may be valid but I don't know. I do think this is working at the moment though for what it's worth.